File: java/src/test/java/com/up42/challenge/controller/FeatureControllerTest.java
package com.up42.challenge.controller;

import com.up42.challenge.service.FeatureService;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.boot.test.mock.mockito.MockBean;
import org.springframework.test.web.servlet.MockMvc;

@WebMvcTest(FeatureController.class)
public class FeatureControllerTest {

    @Autowired
    private MockMvc mvc;

    @MockBean
    private FeatureService service;

    @Test
    void getFeaturesReturnsList() {
        // TODO
    }
}

File: kotlin/src/test/kotlin/com/up42/challenge/controller/FeatureControllerTest.kt
package com.up42.challenge.controller

import com.up42.challenge.service.FeatureService
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
import org.springframework.test.web.servlet.MockMvc
import org.springframework.boot.test.mock.mockito.MockBean

@WebMvcTest(FeatureController::class)
class FeatureControllerTest {

    @Autowired private lateinit var mvc: MockMvc
    @MockBean private lateinit var service: FeatureService

    @Test
    fun `GET features returns list`() {
        // TODO
    }
}